java-manta-client/src/main/java/com/joyent/manta/client/multipart/JobsMultipartManager.java

```diff
@@ -0,0 +1,956 @@
+package com.joyent.manta.client.multipart;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.uuid.Generators;
+import com.joyent.manta.client.MantaClient;
+import com.joyent.manta.client.MantaMetadata;
+import com.joyent.manta.client.MantaObject;
+import com.joyent.manta.client.MantaObjectMapper;
+import com.joyent.manta.client.MantaObjectResponse;
+import com.joyent.manta.client.jobs.MantaJob;
+import com.joyent.manta.client.jobs.MantaJobBuilder;
+import com.joyent.manta.client.jobs.MantaJobPhase;
+import com.joyent.manta.exception.MantaClientHttpResponseException;
+import com.joyent.manta.exception.MantaException;
+import com.joyent.manta.exception.MantaIOException;
+import com.joyent.manta.exception.MantaMultipartException;
+import com.joyent.manta.http.HttpHelper;
+import com.joyent.manta.http.MantaConnectionContext;
+import com.joyent.manta.http.MantaConnectionFactory;
+import com.joyent.manta.http.MantaHttpHeaders;
+import com.joyent.manta.util.MantaUtils;
+import org.apache.commons.lang3.Validate;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPut;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static com.joyent.manta.client.MantaClient.SEPARATOR;
+
+/**
+ * Class providing a jobs-based implementation multipart uploads to Manta.
+ *
+ * @author <a href="https://github.com/dekobon">Elijah Zupancic</a>
+ * @since 2.5.0
```

Comment:
I think we can do `@since version, moved from {@link fully qualified class}` or something like that, since we renamed.